web-registry/src/hooks/useEcocredits.ts

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react';
+
+import { ledgerRESTUri } from '../lib/ledger';
+import { getEcocreditsForAccount } from '../lib/ecocredit';
+import type { BatchInfoWithBalance } from '../types/ledger/ecocredit';
+
+export default function useEcocredits(
```

Comment:
I like this, but something to think about for naming conventions... React hooks conventionally are called `useWhatever`, but in this case we lose some of the context for what the hook _does_. Maybe something like `useQueryEcocredits` or `useGetEcocredits` would make sense? I see you did this in other hooks.


No need to change here, but just a thought.

Reply:
effectively! 

If you notice, the hooks that I am using with `useQuery...` are atomic and correspond to the api offered by our library (`QueryBaskets` => `useQueryBaskets`) (ultimately what they offer are DTOs, the data returned by the API, without modifying). 

On the other hand, one layer above, are the business hooks of our application, which use the original data to aggregate, transform, etc. to prepare the final data needed by the UI components, therefore, these hooks follow a nomenclature `useBasketTokens(Data)`, `useEcocredits(Data)`, etc., referring only to the required data declaratively.

In any case, all this is a first approach, to organize the logic a bit... undoubtedly something that is open, that will change... and of course we will talk shortly, in the coming weeks.

Reply:
ideally, all these hooks could be code generated somehow, and that could be something that some @regen-network/api-react-hooks (just an example name) library could provide, some food for thoughts...

Reply:
I'm already seeing how they are repeated in the next PR. We'll talk about it next week. Super!